system-addon/lib/TopSitesFeed.jsm

```diff
@@ -73,6 +77,13 @@ this.TopSitesFeed = class TopSitesFeed {
           this.refresh(action);
         }
         break;
+      case at.OPEN_NEW_WINDOW:
+        this.openNewWindow(action);
+        break;
+      case at.OPEN_PRIVATE_WINDOW: {
```

Comment:
Oh, I only noticed this when looking at the export to m-c https://reviewboard.mozilla.org/r/137712/diff/1#index_header but these handlers might be more appropriate somewhere outside of `TopSitesFeed`… although `NewTabInit` doesn't seem quite right either. @k88hudson suggestions?